sycl/include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp

```diff
@@ -0,0 +1,274 @@
+#pragma once
+
+#include <CL/sycl/detail/defines_elementary.hpp>
+#include <immintrin.h>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace ext {
+namespace intel {
+namespace experimental::matrix {
+
+enum class matrix_type { a, b, accumulator };
+
+enum class matrix_layout { row_major, col_major, packed };
+
+template <typename Group, typename T, matrix_type MT,
+          size_t Rows = sycl::dynamic_extent,
+          size_t Cols = sycl::dynamic_extent,
+          matrix_layout Layout = matrix_layout::row_major, typename Cond = void>
+struct joint_matrix {
```

Comment:
Please refer to matrix-jit.hpp, we slightly changed the order for the template parameters. Group is now last so we can make it optional. This is necessary because the implementation only supports subgroup scope. this reduces the number of arguments the user has to specify.

Reply:
Thanks, I've switched the Group parameter around.